controllers/reconcile.go

```diff
@@ -992,20 +993,120 @@ func getNavItemPath(feName string, bundleID string, segmentID string) string {
 	return fmt.Sprintf("%s-%s-%s", feName, bundleID, segmentID)
 }
 
-func setupBundlesData(feList *crd.FrontendList, feEnvironment crd.FrontendEnvironment) ([]crd.FrontendBundlesGenerated, []string) {
+type NavSegmentCacheEntry struct {
+	NavItems []crd.ChromeNavItem
+	IsFilled bool
+}
+
+func fillNavRefsTree(navItems []crd.ChromeNavItem, navSegmentsCache map[string]map[string]*NavSegmentCacheEntry, depth uint) ([]crd.ChromeNavItem, error) {
+	parsedNavItems := []crd.ChromeNavItem{}
+	// prevent infinite recursion and navigation nesting
+	// currently max known depth is 2, so 10 should be more than enough for ever
+	// event the depth 2 is challenging when it comes to UX, this should prevent infinite reference loops
+	if depth > 10 {
+		return parsedNavItems, fmt.Errorf("maximum navigation depth reached")
```

Comment:
@Hyperkid123 I think this limitation seems reasonable - though we should write this in our migration guide.

Past that - can we add a test that exercises that this limitation/error is correctly exercised? Otherwise LGTM

Reply:
Oh yeah. I'll add the test.